app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/DeleteRecordController.scala

```diff
@@ -21,27 +21,46 @@ import play.api.mvc.{Action, AnyContent, ControllerComponents}
 import uk.gov.hmrc.http.UpstreamErrorResponse
 import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
 import uk.gov.hmrc.tradergoodsprofilesdatastore.connectors.RouterConnector
-import uk.gov.hmrc.tradergoodsprofilesdatastore.controllers.actions.IdentifierAction
+import uk.gov.hmrc.tradergoodsprofilesdatastore.controllers.actions.{IdentifierAction, StoreLatestAction}
+import uk.gov.hmrc.tradergoodsprofilesdatastore.repositories.RecordsRepository
 
 import javax.inject.Inject
-import scala.concurrent.ExecutionContext
+import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success}
 
 class DeleteRecordController @Inject() (
   cc: ControllerComponents,
   routerConnector: RouterConnector,
-  identify: IdentifierAction
+  recordsRepository: RecordsRepository,
+  identify: IdentifierAction,
+  storeLatestAction: StoreLatestAction
 )(implicit ec: ExecutionContext)
     extends BackendController(cc)
     with Logging {
 
-  def deleteRecord(eori: String, recordId: String): Action[AnyContent] = identify.async { implicit request =>
-    routerConnector.deleteRecord(eori, recordId).map(response => response) transform {
-      case Success(_)                                                             => Success(NoContent)
-      case Failure(cause: UpstreamErrorResponse) if cause.statusCode == NOT_FOUND => Success(NotFound)
-      case Failure(cause: UpstreamErrorResponse)                                  =>
-        logger.error(s"Deleted record failed with ${cause.statusCode} with message: ${cause.message}")
-        Success(InternalServerError)
-    }
+  def deleteRecord(eori: String, recordId: String): Action[AnyContent] = (identify andThen storeLatestAction).async {
+    implicit request =>
+      routerConnector.getRecord(eori, recordId).flatMap { record =>
+        if (record.active) {
```

Comment:
does  B&T returns any inactive record on get single record call, think it returns 400 with error number 26 ?

Reply:
https://jira.tools.tax.service.gov.uk/browse/TGP-1978 :(

Reply:
and also clarified with ben that getting an inactive record return s a 200 and the inactive record